src/js/player.js

```diff
@@ -2379,6 +2393,91 @@ class Player extends Component {
       return this.techGet_('src');
     }
 
+    let src = source;
+
+    if (Array.isArray(source)) {
+      this.cache_.sources = source;
+      src = source[0];
+    } else if (typeof source === 'string') {
+      src = {
+        src: source
+      };
+
+      this.cache_.sources = [src];
+    }
+
+    this.cache_.source = src;
+    this.cache_.src = src.src;
+
+    middleware.setSource(Fn.bind(this, this.setTimeout), src, (src_, mws) => {
+      this.middleware_ = mws;
+
+      const err = this.src_(src_);
+
+      if (err) {
+        if (Array.isArray(source) && source.length > 1) {
+          return this.src(source.slice(1));
+        }
+
+        // We need to wrap this in a timeout to give folks a chance to add error event handlers
+        this.setTimeout(function() {
+          this.error({ code: 4, message: this.localize(this.options_.notSupportedMessage) });
+        }, 0);
+
+        // we could not find an appropriate tech, but let's still notify the delegate that this is it
+        // this needs a better comment about why this is needed
+        this.triggerReady();
+
+        return;
+      }
+
+      middleware.setTech(mws, this.tech_);
+    });
+
+    return this;
+  }
+
+  src_(source) {
+    const sourceTech = this.selectSource([source]);
+
+    if (sourceTech) {
+      if (sourceTech.tech === this.techName_) {
```

Comment:
It would probably be good to have a similar shortout here. 

```js 
if (sourceTech.tech !== this.techName_) {
  // load this technology with the chosen source
  his.loadTech_(sourceTech.tech, sourceTech.source);
  return false;
}
```